src/udev/udev-ctrl.c

```diff
@@ -169,12 +169,6 @@ static int udev_ctrl_connection_event_handler(sd_event_source *s, int fd, uint32
          * To avoid the object freed, let's increment the refcount. */
         uctrl = udev_ctrl_ref(userdata);
 
-        size = next_datagram_size_fd(fd);
-        if (size < 0)
-                return log_error_errno(size, "Failed to get size of message: %m");
-        if (size == 0)
-                return 0; /* Client disconnects? */
-
```

Comment:
Oh, not your fault, but the received size needs to be checked in the below. At least, it must be larger than the size of magic and type.

Reply:
(This is handled in the last commit already)